src/games/strategy/net/ClientLoginHelper.java

```diff
@@ -24,15 +25,18 @@ public boolean login() {
       // write the object output streams magic number
       out.flush();
       final ObjectInputStream in = new ObjectInputStream(streams.getBufferedIn());
-      @SuppressWarnings("rawtypes")
-      final Map challenge = (Map) in.readObject();
+      Object challengeObject = in.readObject();
+      Map<String, String> challenge = null;
+      if(challengeObject instanceof Map){
+        challenge = (Map<String, String>) challengeObject;
+      }
```

Comment:
We really want to remove the casting, put it one spot. This means creating a utility to read a map from an ObjectInputStream. Check for libraries that read and convert objects from ObjectInputStream. The goal is to remove  casting to as few places as possible. `instanceof` checks are a smell that indicate we are not meeting that goal.